tests/unit/projection-test.js

```diff
@@ -1780,6 +1780,257 @@ module('unit/projection', function(hooks) {
     });
   });
 
+  module('Update projection property with resolved value', function(hooks) {
+    // properties for use for initial state
+    const BOOK_ID = 'isbn:9780439708181';
+    // TODO is this valid? we won't have a real ID yeah?
+    const PUBLISHER_ID = 'publisher-abc123';
+    const PUBLISHER_ID_NEW = 'publisher-abc123_new';
+    const PUBLISHER_URN = `urn:${PUBLISHER_CLASS}:${PUBLISHER_ID}`;
+
+    // intial old values
+    const PUBLISHER_NAME = 'MACMILLAN';
+    const PUBLISHER_LOCATION = 'Isle of Arran, Scotland';
+    const PUBLISHER_OWNER = 'Daniel and Alexander Macmillan';
+
+    // properties for use post-patch
+    const NEW_PUBLISHER_NAME = 'MACMILLAN NEW';
+    const NEW_PUBLISHER_LOCATION = 'London, England';
+    const NEW_PUBLISHER_OWNER = 'Holtzbrinck Publishing Group';
+    const NEW_PUBLISHER_URN = `urn:${PUBLISHER_CLASS}:${PUBLISHER_ID_NEW}`;
+
+    hooks.beforeEach(function() {
+      let { store } = this;
+
+      let baseRecord;
+      let projectedExcerpt;
+      let projectedPreview;
+
+      run(() => {
+        baseRecord = store.push({
+          data: {
+            id: BOOK_ID,
+            type: BOOK_CLASS_PATH,
+            attributes: {
+              publisher: PUBLISHER_URN,
+            },
+          },
+          included: [
+            {
+              id: PUBLISHER_ID,
+              type: PUBLISHER_CLASS,
+              attributes: {
+                name: PUBLISHER_NAME,
+                location: PUBLISHER_LOCATION,
+                owner: PUBLISHER_OWNER,
+              },
+            },
+            {
+              id: PUBLISHER_ID,
+              type: PROJECTED_PUBLISHER_CLASS,
+              attributes: {},
+            },
+          ],
+        });
+
+        projectedExcerpt = store.push({
+          data: {
+            id: BOOK_ID,
+            type: BOOK_EXCERPT_PROJECTION_CLASS_PATH,
+            attributes: {},
+          },
+        });
+
+        projectedPreview = store.push({
+          data: {
+            id: BOOK_ID,
+            type: BOOK_PREVIEW_PROJECTION_CLASS_PATH,
+            attributes: {},
+          },
+        });
+      });
+
+      this.records = {
+        baseRecord,
+        projectedExcerpt,
+        projectedPreview,
+      };
+
+      //Adding .setAttribute hook in schema
+      SchemaManager.schema.setAttribute = function(modelName, attr, value, schemaInterface) {
```

Comment:
shouldn't we do this before we push anything into the store?

Reply:
That is a good point.

But Looking at the code of `store.push`, it doesn't invoke `.set` method on the model, as result probably I didn't see any error. 

But I will update the code to set the hook before invoking the store.push.